x/transferinject/keeper/keeper.go

```diff
@@ -0,0 +1,168 @@
+package keeper
+
+import (
+	"errors"
+	"fmt"
+	. "slices"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v6/modules/core/05-port/types"
+
+	rtypes "github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+	"github.com/dymensionxyz/dymension/v3/x/transferinject/types"
+)
+
+type Keeper struct {
+	cdc codec.BinaryCodec
+	porttypes.IBCModule
+	porttypes.ICS4Wrapper
+	rollappKeeper types.RollappKeeper
+	bankKeeper    types.BankKeeper
+}
+
+func NewTransferInject(
+	cdc codec.BinaryCodec,
+	rollappKeeper types.RollappKeeper,
+	bankKeeper types.BankKeeper,
+) *Keeper {
+	return &Keeper{
+		cdc:           cdc,
+		rollappKeeper: rollappKeeper,
+		bankKeeper:    bankKeeper,
+	}
+}
+
+func (t *Keeper) SetMiddleware(
+	ibc porttypes.IBCModule,
+	ics porttypes.ICS4Wrapper,
+) {
+	t.IBCModule = ibc
+	t.ICS4Wrapper = ics
+}
```

Comment:
This keeper is also itself middleware right (or not?)

Can we make it clear in that case that this module and wrapper are the 'next' middleware(s), I suppose

Maybe better to have two setters